core/src/replay_stage.rs

```diff
@@ -974,29 +1010,106 @@ mod test {
     fn test_dead_fork_entry_verification_failure() {
         let keypair1 = Keypair::new();
         let keypair2 = Keypair::new();
-        let res = check_dead_fork(|blockhash, slot| {
+        let res = check_dead_fork(|bank| {
+            let blockhash = bank.last_blockhash();
+            let slot = bank.slot();
             let bad_hash = hash(&[2; 30]);
+            let hashes_per_tick = bank.hashes_per_tick().unwrap_or(0);
             let entry = entry::next_entry(
                 // User wrong blockhash so that the entry causes an entry verification failure
                 &bad_hash,
-                1,
+                hashes_per_tick,
                 vec![system_transaction::transfer_now(
                     &keypair1,
                     &keypair2.pubkey(),
                     2,
-                    *blockhash,
+                    blockhash,
                 )],
             );
             entries_to_test_shreds(vec![entry], slot, slot.saturating_sub(1), false)
         });
 
-        assert_matches!(res, Err(Error::BlobError(BlobError::VerificationFailed)));
+        if let Err(Error::BlockError(block_error)) = res {
+            assert_eq!(block_error, BlockError::InvalidEntryHash);
+        } else {
+            assert!(false);
+        }
+    }
+
+    #[test]
+    fn test_dead_fork_invalid_tick_hash_count() {
+        let res = check_dead_fork(|bank| {
+            let blockhash = bank.last_blockhash();
+            let slot = bank.slot();
+            let hashes_per_tick = bank.hashes_per_tick().unwrap_or(0);
+            assert!(hashes_per_tick > 0);
+
+            let too_few_hashes_tick = Entry::new(&blockhash, hashes_per_tick - 1, vec![]);
+            entries_to_test_shreds(
+                vec![too_few_hashes_tick],
+                slot,
+                slot.saturating_sub(1),
+                false,
+            )
+        });
+
+        if let Err(Error::BlockError(block_error)) = res {
+            assert_eq!(block_error, BlockError::InvalidTickHashCount);
+        } else {
+            assert!(false);
+        }
+    }
+
+    #[test]
+    fn test_dead_fork_invalid_slot_tick_count() {
```

Comment:
@rob-solana in response to this comment: https://github.com/solana-labs/solana/pull/6262#discussion_r336832177...

I have a test here for making sure the fork is marked as dead. Looks like `mark_dead_slot` sets the BankProgress to dead as well as the blocktree slot 👍